include/secp256k1.h

```diff
@@ -638,7 +638,7 @@ SECP256K1_API extern const secp256k1_nonce_function secp256k1_nonce_function_def
  *
  *  Returns: 1: signature created
  *           0: the nonce generation function failed, or the secret key was invalid.
- *  Args:    ctx:       pointer to a context object, initialized for signing.
+ *  Args:    ctx:       pointer to a context object (not secp256k1_context_static).
```

Comment:
Strictly speaking yes. I omitted this because it's perhaps obvious, it adds clutter everywhere, and confuses the user more than it helps: cloning the static context is a pretty obscure operation. (Is there any valid use case even?) I think we could in fact disallow this operation. I doubt this hurts anyone, and it would arguably make the API docs even simpler. But I didn't add this to this PR because it would technically be a backward incompatible change, and its of limited use: We cannot prevent the user from simply copying the memory, e.g., using memcpy.   

Now that you bring this up, what we could do is to add a further comment to the docs of `_context_clone` or the docs of the static context? I think would be the best way to document that the restrictions Do you want me to do this?

Reply:
> Just pointing out that the static context can't be cloned should be sufficient.

"pointing out that the static context can't be cloned" would mean forbidding it. Or do you mean "pointing out that it should not be cloned"?

Reply:
> I think for now it's better to just outlaw cloning the static context, as I think there is no point? 

Agreed. I'm in the middle of doing this but it's a small mess and it should not hold up this PR. I'll open a follow up PR on top of this..